lib/app.rb

```diff
@@ -80,61 +83,60 @@ def run
-          @current_project.values.first << task_name
+          add_task_to_current_project(task_name)
           task_printer.created(task_name)
         when "b"
-          @current_project = false
+          @current_project = {}
           task_printer.double_line
         when "c"
           prompter.new_project_name
-          old_name = @current_project.keys.first
+          old_name = original_name_of_current_project
           new_name = user_input.get_project_or_task_name
-          @current_project[new_name] = @current_project.values.first
-          @current_project.delete(old_name)
+          update_name_association_with_existing_tasks(new_name, old_name)
           project_printer.changed_name(old_name, new_name)
         when "e"
           name = user_input.get_project_or_task_name
-          if (index = @current_project.values.first.find_index(name))
+          if (index = tasks_in_current_project.find_index(name))
             task_printer.editing_task(name)
             prompter.new_task_name
             new_name = user_input.get_project_or_task_name
-            @current_project[@current_project.keys.first][index] = new_name
+            change_name_of_task_in_current_project(index, new_name)
             task_printer.changed_task_name(name, new_name)
           else
             task_printer.does_not_exist(name)
           end
         when "d"
-          project_name = @current_project.keys.first
-          if @current_project.values.first.empty?
+          project_name = original_name_of_current_project
+          if tasks_in_current_project.empty?
             task_printer.no_tasks_created(project_name)
           else
             prompter.existing_task_name
             task_name = user_input.get_project_or_task_name
-            if @current_project[@current_project.keys.first].delete(task_name.strip)
+            if task_can_be_removed_from_current_project?(task_name)
               task_printer.deleted(task_name)
             else
               task_printer.does_not_exist(task_name)
             end
           end
         when "f"
-          project_name = @current_project.keys.first
-          if @current_project.values.first.empty?
+          project_name = original_name_of_current_project
+          if tasks_in_current_project.empty?
             task_printer.no_tasks_created(project_name)
           else
             prompter.existing_task_name
             task_name = user_input.get_project_or_task_name
-            if @current_project[@current_project.keys.first].delete(task_name.strip)
+            if task_can_be_removed_from_current_project?(task_name)
               task_printer.finished(task_name)
             else
               task_printer.does_not_exist(task_name)
             end
           end
         when "ls"
-          if @current_project.values.first.empty?
-            task_printer.no_tasks_created_in_current_project(@current_project)
+          if tasks_in_current_project.empty?
```

Comment:
This one too should move to a method. Just imagine encapsulating every bit of logic as you move it out, and that is your method.

Reply:
✔️